### ISSUE TITLE
Purchase Invoice - error while submitting the purchase invoice

### ISSUE THREAD

**Issue by vinod-8848digital** [8848digital/erpnext#654]
### Information about bug

go to the purchase invoice list
create a new purchase invoice and submit it, then error will raise

https://github.com/user-attachments/assets/83bdebb8-35f9-4370-ac73-d2e53b39fe37

### Module

accounts

### Version

Assets: v0.0.1 (pre-prod)

ERPNext: v15.35.1 (pre-prod)

Frappe Framework: v15.41.0 (pre-prod)

Payments: v0.0.1 (version-15)

Projects: v0.0.1 (master)

sales_commission: v0.0.1 (master)

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

```shell
desk.bundle.MXV7QOYN.js:sourcemap:453 Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 114, in application
    response = frappe.api.handle(request)
  File "apps/frappe/frappe/api/__init__.py", line 49, in handle
    data = endpoint(**arguments)
  File "apps/frappe/frappe/api/v1.py", line 36, in handle_rpc_call
    return frappe.handler.handle()
  File "apps/frappe/frappe/handler.py", line 49, in handle
    data = execute_cmd(cmd)
  File "apps/frappe/frappe/handler.py", line 85, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "apps/frappe/frappe/__init__.py", line 1775, in call
    return fn(*args, **newargs)
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
  File "apps/frappe/frappe/desk/form/save.py", line 37, in savedocs
    doc.submit()
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1048, in submit
    return self._submit()
  File "apps/frappe/frappe/model/document.py", line 1031, in _submit
    return self.save()
  File "apps/frappe/frappe/model/document.py", line 337, in save
    return self._save(*args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 390, in _save
    self.run_post_save_methods()
  File "apps/frappe/frappe/model/document.py", line 1131, in run_post_save_methods
    self.run_method("on_submit")
  File "apps/frappe/frappe/model/document.py", line 962, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1322, in composer
    return composed(self, method, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1304, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "apps/frappe/frappe/model/document.py", line 959, in fn
    return method_object(*args, **kwargs)
  File "apps/erpnext/erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py", line 717, in on_submit
    self.make_gl_entries()
  File "apps/erpnext/erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py", line 754, in make_gl_entries
    make_gl_entries(
  File "apps/erpnext/erpnext/accounts/general_ledger.py", line 40, in make_gl_entries
    create_payment_ledger_entry(
  File "apps/erpnext/erpnext/accounts/utils.py", line 1776, in create_payment_ledger_entry
    ple.submit()
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1048, in submit
    return self._submit()
  File "apps/frappe/frappe/model/document.py", line 1031, in _submit
    return self.save()
  File "apps/frappe/frappe/model/document.py", line 337, in save
    return self._save(*args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 359, in _save
    return self.insert()
  File "apps/frappe/frappe/model/document.py", line 315, in insert
    self.run_post_save_methods()
  File "apps/frappe/frappe/model/document.py", line 1130, in run_post_save_methods
    self.run_method("on_update")
  File "apps/frappe/frappe/model/document.py", line 962, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1322, in composer
    return composed(self, method, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1304, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "apps/frappe/frappe/model/document.py", line 959, in fn
    return method_object(*args, **kwargs)
  File "apps/erpnext/erpnext/accounts/doctype/payment_ledger_entry/payment_ledger_entry.py", line 177, in on_update
    update_voucher_outstanding(
  File "apps/erpnext/erpnext/accounts/utils.py", line 1795, in update_voucher_outstanding
    voucher_outstanding = ple_query.get_voucher_outstandings(vouchers, common_filter=common_filter)
  File "apps/erpnext/erpnext/accounts/utils.py", line 2082, in get_voucher_outstandings
    self.query_for_outstanding()
  File "apps/erpnext/erpnext/accounts/utils.py", line 2045, in query_for_outstanding
    self.voucher_outstandings = self.cte_query_voucher_amount_and_outstanding.run(as_dict=True)
  File "apps/frappe/frappe/query_builder/utils.py", line 87, in execute_query
    result = frappe.db.sql(query, params, *args, **kwargs)  # nosemgrep
  File "apps/frappe/frappe/database/postgres/database.py", line 310, in sql
    return super().sql(modify_query(query), modify_values(values), *args, **kwargs)
  File "apps/frappe/frappe/database/database.py", line 227, in sql
    self._cursor.execute(query, values)
psycopg2.errors.GroupingError: column "tabPayment Ledger Entry.account" must appear in the GROUP BY clause or be used in an aggregate function
LINE 1: WITH vouchers AS (SELECT "account","voucher_type","voucher_n...
```

**Comment by anandk-spyke-o1** [comment]
Fixed in #653
PR : - https://github.com/8848digital/erpnext/pull/658

**Comment by tinadn** [comment]
Resolved. Hence closing.
